app/javascript/packs/admin_users/user.jsx

```diff
@@ -0,0 +1,258 @@
+        // but salary not assigned yet
+        if (data !== null) {
+          this.setState({
+            amount: data.amount,
+            reviewDate: data.review_date,
+            archivedAt: data.archived_at,
+          });
+          this.checkValues();
+        } else {
+          this.setState({
+            amount: 0,
+            reviewDate: 'Not set yet',
+          });
+        }
+      },
+    });
+  }
+
+  setNewSalary() {
+    const today = new Date();
+    const archivedAt = today.toISOString().substring(0, 10);
+    const salary = {
+      amount: this.state.amount,
+      review_date: this.state.reviewDate,
+      archived_at: archivedAt,
+    };
+    this.props.setNewSalary(salary, this.props.user.id);
+  }
+
+  checkValues() {
+    if (this.state.reviewDate === null) {
+      this.setState({ reviewDate: 'Not set yet' });
+    }
+    if (this.state.amount === null) {
+      this.setState({ amount: 0 });
+    }
+  }
+
+
+  handleDelete() {
+    this.props.handleDelete(this.props.user.id);
+  }
+
+  handleEditSalary() {
+    if (this.state.editableSalary) {
+      const salary = {
+        amount: this.state.amount,
+        review_date: this.state.reviewDate,
+        archived_at: this.state.archivedAt,
+      };
+      this.props.handleUpdateSalary(salary, this.props.user.id);
+    }
+    this.setState({ editableSalary: !this.state.editableSalary });
+  }
+
+  handleEdit() {
+    if (this.state.editable) {
+      const user = {
+        id: this.props.user.id,
+        email: this.props.user.email,
+        role: this.state.role,
+        status: this.state.status,
+      };
+      this.props.handleUpdate(user);
+    }
+    this.setState({ editable: !this.state.editable });
+  }
+
+  handleSalaryChange(event) {
+    this.setState({ amount: event.target.value }, () => {
+      this.checkSetSalaryButton();
+    });
+  }
+
+  handleReviewDateChange(event) {
+    this.setState({ reviewDate: event.target.value }, () => {
+      this.checkSetSalaryButton();
+    });
+  }
+
+  checkSetSalaryButton() {
+    if ((this.state.amount > 0) && (/^\d+-\d+-\d+/.test(this.state.reviewDate))) {
+      $('#edit_submit').css('visibility', 'visible');
+    } else {
+      $('#edit_submit').css('visibility', 'hidden');
+    }
+  }
+
+  mouseOverRed() {
+    this.msg.error('You have missed review date!');
+  }
+
+  mouseOverGrey() {
+    this.msg.info('Review salary in few days!');
+  }
+
+  mouseLeave() {
+    this.msg.removeAll();
+  }
+
+  warningIcon() {
+    if ((new Date(this.state.reviewDate) - new Date()) < 0) {
+      return (<div>
+        <img
+          onMouseOver={() => this.mouseOverRed()}
+          onMouseLeave={() => this.mouseLeave()}
+          src="https://cdn2.iconfinder.com/data/icons/freecns-cumulus/32/519791-101_Warning-128.png"
+          alt="red-warn-icon"
+        />
+        <br />
+      </div>);
+    } else if ((new Date(this.state.reviewDate) - new Date()) < 1.21e+9) {
```

Comment:
this is not bad, but there is a good library for handling different time issues - `moment.js`